src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java

```diff
@@ -282,6 +289,20 @@ public void validateCollection(FailureCollector collector) {
     }
   }
 
+  /**
+   * Validates the given database name to consists of characters allowed to represent a dataset.
+   */
+  public void validateDatabaseName(FailureCollector collector) {
+    if (containsMacro(FirestoreConfig.NAME_DATABASE)) {
+      return;
+    }
+
+    if (Strings.isNullOrEmpty(getDatabaseName())) {
```

Comment:
other than checking `null or empty`, please also add validation for database ID based on public doc: https://firebase.google.com/docs/firestore/manage-databases#database_id

Reply:
Done and added tests for each case as well.